components/playlist/browser/resources/components/playlistPlayer.tsx

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { useSelector } from 'react-redux'
+import { RouteComponentProps } from 'react-router-dom'
+
+import * as PlaylistMojo from 'gen/brave/components/playlist/common/mojom/playlist.mojom.m.js'
+import * as States from '../reducers/states'
+import VideoFrame from './videoFrame'
+import PlaylistItem from './playlistItem'
+
+interface MatchParams {
+  playlistId: string
+}
+
+export default function PlaylistPlayer ({
+  match
+}: RouteComponentProps<MatchParams>) {
+  const playlist = useSelector<
```

Comment:
Hey, I see you're using this selector in the Head component too. Did you know you can factor out hooks?

```tsx
const usePlaylist = (id: string) => {
    return useSelector<States.ApplicationState, PlaylistMojo.Playlist | undefined>(state => state.playlistData?.lists.find(e => e.id === id))
}

// then, in your component:
const playlist = usePlaylist(match.params.playlistId)
```